topsort-api-v2.yml

```diff
@@ -629,6 +682,108 @@ components:
             format: double
           minItems: 1
           maxItems: 10000
+ 
+    SponsoredBrandAuctionRequest:
+      type: object
+      description: |
+        Describes the intent of running a sponsored brand auction.
+
+        Exactly **one** of the following fields must be set:
+          * `products`
+          * `category`
+      properties:
+        winners:
+          type: integer
+          format: int32
+          minimum: 1
+          description: Specifies the maximum number of auction winners that should be returned.
+        category:
+          $ref: '#/components/schemas/SingleCategory'
+        products:
+          $ref: '#/components/schemas/Products'
+      required:
+        - winners
+    
+    SponsoredBrandAuctionResult:
+      type: object
+      properties:
+        winners:
+          type: array
+          items:
+            $ref: '#/components/schemas/SponsoredBrandWinner'
+          description: >
+            Array of winner objects in order from highest to lowest bid.
+            It will be empty if there were no qualifying bids or if there was an error.
+        error:
+          type: boolean
+          description: A boolean indicating whether this auction was resolved successfully.
+          example: false
+      required:
+        - winners
+        - error
+
+    SponsoredBrandWinner:
+      type: object
+      required:
+        - rank
+        - productId
+        - resolvedBidId
+        - assets
+      properties:
+        rank:
+          type: integer
+          format: int32
+          description: >
+            Where the product's bid ranked in the auction.
+            One-based, so the product with rank 1 won the auction.
+            In an auction response, the winners array is sorted so rank will match the entry's index.
+          minimum: 1
+        productId:
+          type: string
+          description: |-
+            The marketplace's ID of the winning entity, depending on the target of the campaign.
+          example: p_Mfk15
+        resolvedBidId:
+          type: string
+          description: An opaque Topsort ID to be used when this item is interacted with.
+          example: WyJiX01mazE1IiwiMTJhNTU4MjgtOGVhZC00Mjk5LTgzMjctY2ViYjAwMmEwZmE4IiwibGlzdGluZ3MiLCJkZWZhdWx0IiwiIl0=
+        title:
+          type: string
+          description: An optional title for the sponsored brand selected when creating the campaign.
+        assets:
+          description: Assets used to render the sponsored brand ad.
+          type: array
+          items:
+            $ref: '#/components/schemas/SponsoredBrandAsset'
+          minItems: 1
+
+    SponsoredBrandAsset:
+      type: object
+      properties:
+        url:
+          type: string
+          format: uri
+          description: >
+              A vendor provided asset that the marketplace has to render.
+              The asset will be served by Topsort's CDN.
+        role:
```

Comment:
Should we specify what the valid values are in this case. I think it's only going to be `image` and `logo`.